be/src/Project/Web/Seller/Seller.Web/Areas/Clients/ModelBuilders/ClientFormModelBuilder.cs

```diff
@@ -154,36 +154,36 @@ public async Task<ClientFormViewModel> BuildModelAsync(ComponentModelBase compon
                     if (user is not null)
                     {
                         viewModel.HasAccount = true;
-                    }
 
-                    var approvals = await _approvalsRepository.GetAsync(componentModel.Token, componentModel.Language, null, Constants.DefaultPageIndex, Constants.DefaultItemsPerPage, null);
-                       
-                    if (approvals is not null)
-                    {
-                        var approvalsList = approvals.Data.OrEmptyIfNull().Select(x => new ApprovalViewModel
-                        {
-                            Id = x.Id,
-                            Name = x.Name,
-                            ApprovalDate = null
-                        }).ToList();
+                        var approvals = await _approvalsRepository.GetAsync(componentModel.Token, componentModel.Language, null, Constants.DefaultPageIndex, Constants.DefaultItemsPerPage, null);
```

Comment:
**suggestion:**  Avoid duplicate repository calls, Reduce redundant assignments and minimize nesting 

code suggestion:
```cs
if (approvals is not null)
{
    var userApprovals = await _userApprovalsRepository.GetAsync(
        componentModel.Token,
        componentModel.Language,
        Guid.Parse(user.Id)
    );

    var approvalsList = approvals.Data.OrEmptyIfNull().Select(x => new ApprovalViewModel
    {
        Id = x.Id,
        Name = x.Name,
        ApprovalDate = null,
        IsApproved = false
    }).ToList();

    foreach (var approval in approvalsList)
    {
        var userApproval = userApprovals.FirstOrDefault(x => x.ApprovalId == approval.Id);
        if (userApproval is not null)
        {
            approval.IsApproved = true;
            approval.ApprovalDate = userApproval.CreatedDate;
        }
    }

    viewModel.ClientApprovals = approvalsList;
}

```